### ISSUE TITLE
Shorthand syntax for single-declaration attribute sets

### ISSUE THREAD

**Issue by shlevy** [NixOS/nix#7219]
Would be nice if

```nix
{
  foo = bar = baz;
  qux = inherit (quux) corge grault;
}
```

were shorthand for

```nix
{
  foo = { bar = baz; };
  qux = { inherit (quux) corge grault; };
}
```

Am willing to implement if it's wanted.

**Comment by Radvendii** [comment]
the first one is already
```
{
  foo.bar = baz;
}
```
and I personally find `foo = bar = baz` syntax confusing. I'm sure I would get used to it, but it reads like `C` syntax for assigning `foo` and `bar` both the value `baz`.

The second one seems convenient, but I wonder if it's worth the confusion of more syntax. I'm on the fence because I've encountered this situation a few times and wanted a nicer syntax for it, but I haven't encountered it *that* much, and the existing syntax is... fine?

**Comment by tomberek** [comment]
This does not seem that much more ergonomic, not more clear. Is there a common pattern that motivates this?

**Comment by shlevy** [comment]
`foo.bar = baz;` only works in defining an attribute set, I intended this to be an example illustrating any expression context, e.g. `let ... in bar = baz`.

@tomberek Motivation is just noticing it would be nice enough times to propose it.

**Comment by thufschmitt** [comment]
I agree with the comments, it's quite surprising as a syntax. Esp. the `inherit` one (at first read my brain just ignored the `qux =` part and assumed that was just a plain `inherit` as we already have it). And given that it's just a small nicety I wouldn't favor it.

**Comment by tomberek** [comment]
Closing as stale. Re-open if needed.
